hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/factories/OutboundResponseFromHubToResponseTransformerFactory.java

```diff
@@ -49,4 +54,8 @@ public OutboundResponseFromHubToResponseTransformerFactory(OutboundLegacyRespons
         }
         return simpleProfileOutboundResponseFromHubToResponseTransformerProvider.get();
     }
+
+    public Function<AuthnResponseFromCountryContainerDto, String> getCountryTransformer() {
```

Comment:
I've delved into the difference between a "legacy" response and a "samlProfile" response. The only difference is the status marshaller that they use. The "samlProfile" status marshaller will map a `NoMatchingServiceMatchFromHub` status slightly differently. In the non-matching flow being implemented here, the status is being hardcoded into the response as we already know we're in a success journey. See `SamlProfileTransactionIdaStatusMarshaller` and `TransactionIdaStatusMarshaller` for the implementation.

As for the SHA1/SHA256 switch, I've checked and all services have been upgraded to using SHA256. We could probably remove the extra call to config...

Finally, the only services that have `hubShouldSignResponseMessages` set to false in config are test-rp, and for some reason `dvsa-declaration` in integration (but not prod).

I think that means it's safe to skip the config calls in this flow and just return the new function here...